pkg/rancher-desktop/main/deploymentProfiles.ts

```diff
@@ -48,19 +49,33 @@ export function readDeploymentProfiles(): settings.DeploymentProfileType {
     for (const key of [nativeReg.HKLM, nativeReg.HKCU]) {
       const registryKey = nativeReg.openKey(key, REGISTRY_PATH_PROFILE.join('\\'), nativeReg.Access.READ);
 
+      if (!registryKey) {
+        continue;
+      }
+      const defaultsKey = nativeReg.openKey(registryKey, 'Defaults', nativeReg.Access.READ);
+      const lockedKey = nativeReg.openKey(registryKey, 'Locked', nativeReg.Access.READ);
+
       try {
-        if (registryKey !== null) {
-          profiles.defaults = readRegistryUsingSchema(settings.defaultSettings, registryKey, ['Defaults']);
-          profiles.locked = readRegistryUsingSchema(settings.defaultSettings, registryKey, ['Locked']);
+        if (defaultsKey) {
+          profiles.defaults = readRegistryUsingSchema(settings.defaultSettings, defaultsKey) ?? {};
+        }
+        if (lockedKey) {
+          profiles.locked = readRegistryUsingSchema(settings.defaultSettings, lockedKey) ?? {};
         }
       } catch (err) {
         console.error( `Error reading deployment profile: ${ err }`);
       } finally {
-        if (typeof registryKey !== 'undefined') {
-          nativeReg.closeKey(registryKey);
+        nativeReg.closeKey(registryKey);
+        if (defaultsKey) {
+          nativeReg.closeKey(defaultsKey);
+        }
+        if (lockedKey) {
+          nativeReg.closeKey(lockedKey);
         }
       }
-      if (typeof profiles.defaults !== 'undefined' || typeof profiles.locked !== 'undefined') {
+      // If we found something in the HKLM Defaults or Locked registry hive, don't look at the user's
+      // Alternatively, if the keys work, we could break, even if both hives are empty.
```

Comment:
Actually, `parseMultiString` does ok for both single-string values and DWORDs

Reply:
Oh, that's surprising given its implementation. It's just calling `toString` on a `Buffer` that it would otherwise use `readUInt32LE` etc on?
https://github.com/simonbuchan/native-reg/blob/84faae6bc023c6a5d89adf77c176ddc99f6d1f45/src/index.ts#L308-L312

Reply:
Moved to issue #4153